common/src/main/java/com/wynntils/models/containers/BankModel.java

```diff
@@ -0,0 +1,156 @@
+
+    public BankModel(ContainerModel containerModel) {
+        super(List.of(containerModel));
+    }
+
+    @SubscribeEvent
+    public void onScreenInit(ScreenInitEvent e) {
+        if (!(e.getScreen() instanceof AbstractContainerScreen<?> screen)) return;
+
+        if (Models.Container.isBankScreen(screen)) {
+            currentContainer = SearchableContainerType.BANK;
+        } else if (Models.Container.isBlockBankScreen(screen)) {
+            currentContainer = SearchableContainerType.BLOCK_BANK;
+        } else if (Models.Container.isBookshelfScreen(screen)) {
+            currentContainer = SearchableContainerType.BOOKSHELF;
+        } else if (Models.Container.isMiscBucketScreen(screen)) {
+            currentContainer = SearchableContainerType.MISC_BUCKET;
+        } else {
+            currentContainer = null;
+            currentPage = 1;
+            return;
+        }
+
+        currentPage = Models.Container.getCurrentBankPage(screen);
+
+        editingName = false;
+    }
+
+    @SubscribeEvent
+    public void onScreenClose(ScreenClosedEvent e) {
+        currentContainer = null;
+        currentPage = 1;
+        editingName = false;
+    }
+
+    public String getPageName(int page) {
+        return switch (currentContainer) {
+            case BANK -> customBankPageNames.get().getOrDefault(page, "");
+            case BLOCK_BANK -> customBlockBankPageNames.get().getOrDefault(page, "");
+            case BOOKSHELF -> customBookshelfPageNames.get().getOrDefault(page, "");
+            case MISC_BUCKET -> customMiscBucketPageNames.get().getOrDefault(page, "");
+            default -> "";
+        };
+    }
+
+    public void saveCurrentPageName(String nameToSet) {
+        switch (currentContainer) {
+            case BANK -> {
+                customBankPageNames.get().put(currentPage, nameToSet);
+            }
+            case BLOCK_BANK -> {
+                customBlockBankPageNames.get().put(currentPage, nameToSet);
+            }
+            case BOOKSHELF -> {
+                customBookshelfPageNames.get().put(currentPage, nameToSet);
+            }
+            case MISC_BUCKET -> {
+                customMiscBucketPageNames.get().put(currentPage, nameToSet);
+            }
+        }
+
+        editingName = false;
+    }
+
+    public int getFinalPage() {
+        return switch (currentContainer) {
+            case BANK -> finalBankPage.get();
+            case BLOCK_BANK -> finalBlockBankPage.get();
+            case BOOKSHELF -> finalBookshelfPage.get();
+            case MISC_BUCKET -> finalMiscBucketPage.get();
+            default -> 1;
+        };
+    }
+
+    public void updateFinalPage() {
+        switch (currentContainer) {
+            case BANK -> {
+                finalBankPage.store(currentPage);
+            }
+            case BLOCK_BANK -> {
+                if (currentPage > finalBlockBankPage.get()) {
+                    finalBlockBankPage.store(currentPage);
+                }
```

Comment:
Why the special case?

Reply:
Because block bank has different pages for your personal storage and each island storage. So don't want you having all 12 pages in personal storage, then swapping to island storage and it resetting back to page 1 as final.

When I get around to making the Island Model so we can detect which island we're on this can hopefully be handled better.

Reply:
Okay. sounds good.